File: Java/Program/Hdd.java
// Hdd.java
public class Hdd extends Storage {
    private int rpm;        // Rotations per minute
    private int bufferMB;   // Buffer memory in MB
    
    public Hdd() {}
    
    public Hdd(String merk, String nama, int kapasitasGB, int kecepatanMBps, String interfaceType, int rpm, int bufferMB) {
        super(merk, nama, kapasitasGB, kecepatanMBps, interfaceType);
        this.rpm = rpm;
        this.bufferMB = bufferMB;
    }
    
    // Setter methods
    public void setRpm(int rpm) {this.rpm = rpm;}
    public void setBufferMB(int bufferMB) {this.bufferMB = bufferMB;}
    
    // Getter methods
    public int getRpm() {return rpm;}
    public int getBufferMB() {return bufferMB;}
    
    @Override
    public void displayInfo() {
        System.out.println(getNama() + " (" + getKapasitasGB() + " GB, Interface: " + getConnectionType() + ") " + 
                          "kecepatan: " + getKecepatanMBps() + "MBps" + " RPM: " + rpm + " Buffer: " + bufferMB + "MB");
    }
    
    
}

File: Python/program/Hdd.py
from Storage import Storage

class Hdd(Storage):
    def __init__(self, merek, nama, kapasitasGB, kecepatanMBps, interfaceType, rpm, bufferMB):
        super().__init__(merek, nama, kapasitasGB, kecepatanMBps, interfaceType)
        self.rpm = rpm
        self.bufferMB = bufferMB

    def setRpm(self, rpm):
        self.rpm = rpm
    
    def setBufferMB(self, bufferMB):
        self.bufferMB = bufferMB
    
    def getRpm(self):
        return self.rpm
    
    def getBufferMB(self):
        return self.bufferMB

    def getConnectionType(self):
        return self.interfaceType

    def displayInfo(self):
         print(f"{self.nama} ({self.kapasitasGB} GB, Interface: {self.getConnectionType()}) kecepatan: {self.kecepatanMBps} MBps RPM: {self.rpm} Buffer: {self.bufferMB} MB")